src/lib/KeepaliveManager.ts

```diff
@@ -88,9 +91,11 @@ export default class KeepaliveManager {
 		this.counter = 0
 
 		// https://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Figure_3.5_Keep
+		// @TODO: this._keepalive may be undefined. This operation will result in NaN.
 		const keepAliveTimeout = Math.ceil(this._keepalive * 1.5)
 
 		this._keepaliveTimeoutTimestamp = Date.now() + keepAliveTimeout
+		// @TODO: this._keepalive may be undefined. This operation will result in NaN.
 		this._intervalEvery = Math.ceil(this._keepalive / 2)
```

Comment:
A dangerous code that was revealed by using a strict TypeScript configuration. You have two very clear cases where those properties can unexpectedly end up as `NaN`.